Smt/Query.lean

```diff
@@ -250,21 +250,21 @@ def addCommand (cmd : Command) (cmds : List Command) : MetaM (List Command) := d
   | _ => pure ()
   return cmds
 
-def emitVertex (cmds : Std.HashMap Expr Command) (e : Expr) : StateT (List Command) MetaM Unit := do
-  trace[smt.debug.query] "emitting {e}"
+def emitVertex (cmds : HashMap Expr Command) (e : Expr) : StateT (List Command) MetaM Unit := do
+  trace[smt.debug.translate.query] "emitting {e}"
   let some cmd := cmds.find? e | throwError "no command was computed for {e}"
   set (← addCommand cmd (← get))
 
 def generateQuery (goal : Expr) (hs : List Expr) : MetaM (List Command) :=
-  withTraceNode `smt.debug.query (fun _ => pure .nil) do
-    trace[smt.debug.query] "Goal: {← inferType goal}"
-    trace[smt.debug.query] "Provided Hints: {hs}"
+  withTraceNode `smt.debug.translate.query (fun _ => pure .nil) do
+    trace[smt.debug.translate.query] "Goal: {← inferType goal}"
+    trace[smt.debug.translate.query] "Provided Hints: {hs}"
     let ((_, st), _) ← QueryBuilderM.buildDependencyGraph goal
       |>.run { toDefine := hs : QueryBuilderM.Config }
       |>.run { : QueryBuilderM.State }
       |>.run { : TranslationM.State }
-    trace[smt.debug.query] "Dependency Graph: {st.graph}"
-    let (_, cmds) ← StateT.run (st.graph.dfs $ emitVertex st.commands) []
+    trace[smt.debug.translate.query] "Dependency Graph: {st.graph}"
+    let (_, cmds) ← StateT.run (st.graph.orderedDfs (hs ++ [goal]) (emitVertex st.commands)) []
```

Comment:
Could you write a comment explaining why we need to traverse the graph in this particular order rather than the `g.vertices` order as before?

Reply:
Done!